### ISSUE TITLE
DNA-mapping 2.0.1: badly delimited number splits into two tokens

### ISSUE THREAD

**Issue by AdrijaK** [maxplanck-ie/snakepipes#588]
I suspect this error is due to the $PWD containing `02_` 

```bash
echo $PWD
```

```
/localhome/inst/usr123/analysis/02_MLL/data/chip-seq
```

setup snakepipes using `00_setup_snakePipes.txt` (attached)

```bash
preprocessing -i $PWD/rawfastq -o $PWD/H3K27ac --sampleSheet $PWD/H3K27ac_test.tsv --clumpifyOptions 'spany=t adjacent=t' --optDedupDist 40 --fastqc --local --jobs 40

DNA-mapping -i "$PWD/H3K27ac" -o "$PWD/H3K27ac" --local --trim --fastqc --plotFormat pdf --dedup --mapq 3 --DAG --verbose --jobs 60 GRCm38_M24_basic
```

```
---- This analysis has been done using snakePipes version 2.0.1 ----
SystemExit in line 40 of /localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/workflows/DNA-mapping/internals.snakefile:
1
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/workflows/DNA-mapping/Snakefile", line 18, in <module>
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/workflows/DNA-mapping/internals.snakefile", line 40, in <module>
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/_sitebuiltins.py", line 26, in __call__
Error: <stdin>: syntax error in line 2 near 'Error'
Warning: syntax ambiguity - badly delimited number '02_' in line 2 of <stdin> splits into two tokens
Traceback (most recent call last):
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/bin/DNA-mapping", line 173, in <module>
    main()
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/bin/DNA-mapping", line 165, in main
    snakemake_cmd = cf.commonYAMLandLogs(baseDir, workflowDir, defaults, args, __file__)
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/common_functions.py", line 515, in commonYAMLandLogs
    subprocess.check_call("dot -Tpdf -o{}/{}_pipeline.pdf".format(args.outdir, workflowName), stdin=DAGproc.stdout, shell=True)
  File "/localhome/inst/usr123/analysis/software/anaconda3/envs/snakePipes/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
```

sample sheet for preprocessing:
```bash
cat H3K27ac_test.tsv
```

```H3K27ac-dox-1_S2_L001_R1_001.fastq.gz   _R1     H3K27ac-dox-rep1
H3K27ac-dox-1_S2_L002_R1_001.fastq.gz   _R1     H3K27ac-dox-rep1
H3K27ac-dox-1_S2_L003_R1_001.fastq.gz   _R1     H3K27ac-dox-rep1
H3K27ac-dox-1_S2_L004_R1_001.fastq.gz   _R1     H3K27ac-dox-rep1
H3K27ac-dox-3_S8_L001_R1_001.fastq.gz   _R1     H3K27ac-dox-rep3
H3K27ac-dox-3_S8_L002_R1_001.fastq.gz   _R1     H3K27ac-dox-rep3
H3K27ac-dox-3_S8_L003_R1_001.fastq.gz   _R1     H3K27ac-dox-rep3
H3K27ac-dox-3_S8_L004_R1_001.fastq.gz   _R1     H3K27ac-dox-rep3
H3K27ac-veh-1_S1_L001_R1_001.fastq.gz   _R1     H3K27ac-veh-rep1
H3K27ac-veh-1_S1_L002_R1_001.fastq.gz   _R1     H3K27ac-veh-rep1
H3K27ac-veh-1_S1_L003_R1_001.fastq.gz   _R1     H3K27ac-veh-rep1
H3K27ac-veh-1_S1_L004_R1_001.fastq.gz   _R1     H3K27ac-veh-rep1
H3K27ac-veh-3_S3_L001_R1_001.fastq.gz   _R1     H3K27ac-veh-rep3
H3K27ac-veh-3_S3_L002_R1_001.fastq.gz   _R1     H3K27ac-veh-rep3
H3K27ac-veh-3_S3_L003_R1_001.fastq.gz   _R1     H3K27ac-veh-rep3
H3K27ac-veh-3_S3_L004_R1_001.fastq.gz   _R1     H3K27ac-veh-rep3
FA-dox_S8_L001_R1_001.fastq.gz  _R1     input-dox-rep1
FA-dox_S8_L002_R1_001.fastq.gz  _R1     input-dox-rep1
FA-dox_S8_L003_R1_001.fastq.gz  _R1     input-dox-rep1
FA-dox_S8_L004_R1_001.fastq.gz  _R1     input-dox-rep1
FA-veh_S3_L001_R1_001.fastq.gz  _R1     input-veh-rep1
FA-veh_S3_L002_R1_001.fastq.gz  _R1     input-veh-rep1
FA-veh_S3_L003_R1_001.fastq.gz  _R1     input-veh-rep1
FA-veh_S3_L004_R1_001.fastq.gz  _R1     input-veh-rep1
```

[00_setup_snakePipes.txt](https://github.com/maxplanck-ie/snakepipes/files/4317092/00_setup_snakePipes.txt)

**Comment by katsikora** [comment]
Hi Adrija,

I can reproduce your error. It disappears when you remove `--DAG`. This looks like a minor issue on our side, we'll fix it. You can in the meantime run your workflow without this option, this just produces a graphical representation of the rules connections in the pipeline. 

Also, please note that you'll either have to rename your files or update the read dictionary, as there is currently a mismatch:
your files are named like
sample + _R1_001 + .fastq.gz 
the default that the pipeline expects:
sample + _R1 + .fastq.gz

**Comment by AdrijaK** [comment]
Dear Katarzyna, 
Thank you for the advice, I will follow the suggestions.

**Comment by AdrijaK** [comment]
I am a bit confused about the read  naming: 

I the [preprocessing documentation](https://snakepipes.readthedocs.io/en/latest/content/workflows/preprocessing.html#preprocessing)

the samples are names with _R1_001.fastq.gz and their reads are "_R1" and "_R2". 

```
sample1_S1_L001_R1_001.fastq.gz _R1     sample1
sample1_S1_L001_R2_001.fastq.gz _R2     sample1
sample1_S1_L002_R1_001.fastq.gz _R1     sample1
sample1_S1_L002_R2_001.fastq.gz _R2     sample1
sample1_S1_L003_R1_001.fastq.gz _R1     sample1
sample1_S1_L003_R2_001.fastq.gz _R2     sample1
```

Doesnt the pipeline take care of the _001 after the read symbols?

**Comment by dpryan79** [comment]
Once the preprocessing pipeline is done, the resulting files won't have 001.fastq.gz in them, they'll be `sample1_R1.fastq.gz` and `sample1_R2.fastq.gz`.

**Comment by AdrijaK** [comment]
So that means I don't have to rename my files, right? 

> Also, please note that you'll either have to rename your files or update the read dictionary, as there is currently a mismatch:
your files are named like
sample + _R1_001 + .fastq.gz
the default that the pipeline expects:
sample + _R1 + .fastq.gz

**Comment by dpryan79** [comment]
Correct, once you've run the preprocessing pipeline you won't need to rename anything, that will already have been done.

**Comment by AdrijaK** [comment]
Perfect, thanks.

**Comment by katsikora** [comment]
The DAG issue is now fixed in develop PR#590
